VisualRiders.PointOfSale.Project/Controllers/EmployeesController.cs

```
@@ -0,0 +1,102 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using VisualRiders.PointOfSale.Project.Models;
+using VisualRiders.PointOfSale.Project.Repositories;
+
+namespace VisualRiders.PointOfSale.Project.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class EmployeesController : Controller
+    {
+        private readonly EmployeesRepository _employeesRepository;
+
+        public EmployeesController(EmployeesRepository employeesRepository)
+        {
+            _employeesRepository = employeesRepository;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(201)]
+        public ActionResult<Employee> Create(Employee employee)
+        {
+            _employeesRepository.Create(employee);
+
+            return CreatedAtAction("GetById", new { id = employee.Id }, employee);
+        }
+
+        [HttpGet]
+        public List<Employee> GetAll()
+        {
+            return _employeesRepository.GetAll();
+        }
+
+        [HttpGet("{id:guid}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(404)]
+        public ActionResult<Employee> GetById(Guid id)
+        {
+            var employee = _employeesRepository.GetById(id);
+
+            if (employee == null)
+            {
+                return NotFound();
+            }
+
+            return employee;
+        }
+
+
+        [HttpPut("{id:guid}")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public ActionResult UpdateById(Employee employee)
```

We already talked that the implementation where you replace the entire entity is not ideal. You are not only blindly replacing fields which may need some logic but are also swallowing the error when an entity with the requested ID cannot be found and the update fails.

---

Fixed. Probably implementation will change when DTO will be added.